### DISCUSSION TITLE
posts/2022/06/get-good-git-info-hugo/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#25] [Announcements]
# posts/2022/06/get-good-git-info-hugo/

It’s easy to fetch and display your Hugo repo’s Git data.

https://www.brycewray.com/posts/2022/06/get-good-git-info-hugo/

**Comment by rodrigoalcarazdelaosa** [comment]
Hey, Bryce, I've borrowed this code for [my site](https://physichemically.com), but I'd like to show it on my footer. The thing is that I'd like to show the very latest commit of the repo everytime, but with this I get the latest commit where that specific page (file) was modified, which changes from page to page. How can we get the very latest commit info, independent from the page we're navigating? Thanks in advance!

**Reply by brycewray** [reply]
@rodrigoalcarazdelaosa I think in that case you’d dispense with the `fetch-depth` setting. In my own case, having that setting was the only way to **prevent** what you’re requesting. 😄 However, I’m using a GitHub Action, so YMMV. Hope that helps.

**Reply by rodrigoalcarazdelaosa** [reply]
Thanks, Bryce. I’m deploying directly with Netlify so I’m not sure I can easily set that `fetch-depth` 🤔. I’ll keep investigating…

**Reply by brycewray** [reply]
> Thanks, Bryce. I’m deploying directly with Netlify so I’m not sure I can easily set that `fetch-depth` 🤔. I’ll keep investigating…

@rodrigoalcarazdelaosa I actually was under the impression that, unless you **could** specify it, you’d get your desired result by default, but perhaps that’s only for a CI/CD operation like that of GHA or GitLab CI. If you find the answer, let me know and I’ll update this post to include it.

**Reply by brycewray** [reply]
@rodrigoalcarazdelaosa This may also be of help, since it appears this person wanted the same result you do:

https://github.com/gohugoio/hugo/issues/9738

**Reply by rodrigoalcarazdelaosa** [reply]
Thanks, Bryce. The solution proposed on that issue will work but complicates the build process. I hope @bep includes soon a new variable to access the global latest commit with the known syntax we have now...

**Reply by rodrigoalcarazdelaosa** [reply]
I ended up implementing [the solution proposed by @jmooring](https://github.com/gohugoio/hugo/issues/9738#issuecomment-1086669372) so now I can have a link to my very latest commit [on my footer](https://physichemically.com), with this line:
```
{{ with site.Params.git.last_commit }}Latest&nbsp;commit:&nbsp;<a href="https://github.com/rodrigoalcarazdelaosa/fisiquimicamente/commit/{{ .hash }}" target="_blank" rel="noopener">{{ .abbrev_hash }}</a>{{ end }}
```
<img width="757" alt="Captura de Pantalla 2022-10-07 a las 9 59 16" src="https://user-images.githubusercontent.com/9049533/194503174-a4991b62-461a-4f08-a233-4af473cc9ee8.png">

and also a link to the last edit of [each page](https://physichemically.com/blog/2020/04/28/hello-world/), using:
```
{{- if $page.GitInfo -}}{{ i18n "last_edit" }}<a href="https://github.com/rodrigoalcarazdelaosa/fisiquimicamente/commit/{{ $page.GitInfo.Hash }}" target="_blank" rel="noopener">{{ $page.GitInfo.AbbreviatedHash }}</a>{{- end -}}
```
<img width="1031" alt="Captura de Pantalla 2022-10-07 a las 9 58 01" src="https://user-images.githubusercontent.com/9049533/194502887-34b45b45-8cf7-4aba-b6b0-0d3c6afebea6.png">

**Reply by brycewray** [reply]
@rodrigoalcarazdelaosa I have now added this info to the end of the original piece (and its version on dev.to). Thanks for bringing it up in the first place!
